dev/Deployment/DeploymentManager.cpp

```diff
@@ -0,0 +1,249 @@
+        {
+            return status;
+        }
+
+        std::wstring frameworkPackageFullName{ packageFullName };
+        auto initializeResult{ DeployPackages(frameworkPackageFullName) };
+        auto initializeStatus{ winrt::make<implementation::DeploymentStatus>(SUCCEEDED(initializeResult), FAILED(initializeResult), initializeResult) };
+        return initializeStatus;
+    }
+
+    MddCore::PackageInfo DeploymentManager::GetPackageInfoForPackage(std::wstring const& packageFullName)
+    {
+        wil::unique_package_info_reference packageInfoReference;
+        THROW_IF_WIN32_ERROR(OpenPackageInfoByFullName(packageFullName.c_str(), 0, &packageInfoReference));
+        return MddCore::PackageInfo::FromPackageInfoReference(packageInfoReference.get());
+    }
+
+    // Borrowed and repurposed from Dynamic Dependencies
+    std::vector<std::wstring> DeploymentManager::FindPackagesByFamily(std::wstring const& packageFamilyName)
+    {
+        UINT32 count{};
+        UINT32 bufferLength{};
+        const auto rc{ FindPackagesByPackageFamily(packageFamilyName.c_str(), PACKAGE_FILTER_HEAD | PACKAGE_FILTER_DIRECT, &count, nullptr, &bufferLength, nullptr, nullptr) };
+        if (rc == ERROR_SUCCESS)
+        {
+            // The package family has no packages registered to the user
+            return std::vector<std::wstring>();
+        }
+        else if (rc != ERROR_INSUFFICIENT_BUFFER)
+        {
+            THROW_WIN32(rc);
+        }
+
+        auto packageFullNames{ wil::make_unique_cotaskmem<PWSTR[]>(count) };
+        auto buffer{ wil::make_unique_cotaskmem<WCHAR[]>(bufferLength) };
+        THROW_IF_WIN32_ERROR(FindPackagesByPackageFamily(packageFamilyName.c_str(), PACKAGE_FILTER_HEAD | PACKAGE_FILTER_DIRECT, &count, packageFullNames.get(), &bufferLength, buffer.get(), nullptr));
+
+        std::vector<std::wstring> packageFullNamesList;
+        for (UINT32 index=0; index < count; ++index)
+        {
+            const auto packageFullName{ packageFullNames[index] };
+            packageFullNamesList.push_back(std::wstring(packageFullName));
+        }
+        return packageFullNamesList;
+    }
+
+    HRESULT DeploymentManager::VerifyPackage(const std::wstring& packageFamilyName, const PACKAGE_VERSION targetVersion) try
+    {
+        auto packageFullNames{ FindPackagesByFamily(packageFamilyName) };
+        bool match{ false };
+        for (const auto& packageFullName : packageFullNames)
+        {
+            auto packagePath{ GetPackagePath(packageFullName) };
+            if (packagePath.length() == 0)
+            {
+                continue;
+            }
+
+            auto packageId{ MddCore::PackageId::FromPackageFullName(packageFullName.c_str()) };
+            if (packageId.Version().Version >= targetVersion.Version)
+            {
+                match = true;
+                break;
+            }
+        }
+
+        if (!match)
+        {
+            return HRESULT_FROM_WIN32(ERROR_NOT_FOUND);
```

Comment:
Would you want logging here? If so `RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), !match);` ?